package/base-files/files/lib/functions/uci-defaults.sh

```diff
@@ -617,6 +617,52 @@ ucidef_set_hostname() {
 	json_select ..
 }
 
+ucidef_set_wireless() {
+	local ssid="$1"
+	local encryption="$2"
+	local key="$3"
+
+	json_select_object wireless
+		json_add_string ssid "$ssid"
+		[ -n "$encryption" ] && json_add_string country "$encryption"
+		[ -n "$key" ] && json_add_string key "$key"
+	json_select ..
+}
+
+ucidef_enable_wireless() {
+	local bands="$1"
+	local enable_2ghz
+	local enable_5ghz
+
+	case "$bands" in
+		*2*|*b*)
+			enable_2ghz=1
+			;;
+		all)
+			enable_2ghz=1
+			enable_5ghz=1
+			;;
+	esac
+
+	case "$bands" in
+		*5*|*a*)
+			enable_5ghz=1
+			;;
+	esac
```

Comment:
Is there a reason this is split into two *case* statements?
Unless I'm missing something (which is possible), it seem only one is required.

Reply:
It's useful to have it in two statements, so if both bands are explicitly mentioned, ie. "ab", "ba", "a,b", "5g2g", "2g+5g" or anything like that the order doesn't matter.
I'm expecting we are going to add additional bands for 6 GHz and 60 GHz bands at some point, so I didn't want to fix the exact syntax expected by the ucidef functions to be to strict at this point, as we will anyway have to waste some more thought on this...

Reply:
That makes sense to me, thanks.
I got confused as having "ab" and similar was not mentioned on the description.
In fact, that related to my next question (adding it on a different comment).